### ISSUE TITLE
Support code folding ribbon

### ISSUE THREAD

**Issue by cjwcommuny** [kilbd/nova-rust#46]
**Is your feature request related to a problem? Please describe.**

Currently the extension doesn't support folding ribbon as shown in the screenshot.

<img width="1112" alt="截屏2023-06-23 18 10 57" src="https://github.com/kilbd/nova-rust/assets/23581749/264d653c-3e72-4357-a5e2-7b989a7e7cd7">

**Describe the solution you'd like**

Support it!

**Additional context**

Environment:

- Nova 11.2
- nova-rust 2.3.2

**Comment by kilbd** [comment]
Yep, I should add code folding queries. Thanks for bringing this to my attention! I feel like folds used to come for free with symbols but maybe that changed with Tree Sitter? Admittedly, I don't use folding because folds are forgotten between sessions.

**Comment by kilbd** [comment]
I'm not crazy – after commenting out Tree Sitter so it fell back on regex syntax definitions, fold ribbons appeared:

<img width="788" alt="Screenshot 2023-06-25 at 10 37 12 AM" src="https://github.com/kilbd/nova-rust/assets/2053455/ed75d16e-fa72-44aa-a0a0-1864eca88d19">


Good to know!
